### PR TITLE
Update screen recorder for tests of android_client, to make sure the screen recording is performed correctly

### PR DESCRIPTION
**Pull request by olivershen-wow**
Update screen recorder for tests of android_client, to make sure the screen recording is performed correctly.

**Linked GitHub issue ID**: #  

## Pull Request Checklist

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Code compiles correctly with all tests are passed.
- [ ] I've read the [contributing guide](https://github.com/microsoft/HydraLab/blob/main/CONTRIBUTING.md#making-changes-to-the-code) and followed the recommended practices.
- [x] [Wikis](https://github.com/microsoft/HydraLab/wiki) or [README](https://github.com/microsoft/HydraLab/blob/main/README.md) have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?
*If this introduces a breaking change for Hydra Lab users, please describe the impact and migration path.*

- [ ] Yes
- [x] No

## Pull Request Description

**A few words to explain your changes**:  

### How you tested it
*Please make sure the change is tested, you can test it by adding UTs, do local test and share the screenshots, etc.*
- Update screen recorder for tests of android_client
- Add comment for current device action definition
- Abstract retry logic into ADBOperateUtil.pullFileToDir

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Technical design
- [ ] Build related changes
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Code style update (formatting, renaming) or Documentation content changes
- [ ] Other (please describe): 

### Feature UI screenshots or Technical design diagrams
*If this is a relatively large or complex change, kick it off by drawing the tech design with PlantUML and explaining why you chose the solution you did and what alternatives you considered, etc...*

### CODE REVIEW

File: common/src/main/java/com/microsoft/hydralab/common/screen/ADBScreenRecorder.java [hunk 1]
```diff
@@ -93,12 +93,14 @@ public void startRecord(int maxTimeInSecond) {
...
                     }
                     deviceInfo.finishCommand();
+                    // make sure the recording procedure is stopped completely
+                    ThreadUtil.safeSleep(2000);
 
                     String outputFilePrefix = new File(baseFolder, DateUtil.fileNameDateDashFormat.format(new Date())).getAbsolutePath();
-
                     final String outFileFullPath = outputFilePrefix + "_" + totalTime + "_" + (totalTime + timeSpan) + ".mp4";
                     String pullComm = String.format("pull %s %s", fileName, outFileFullPath);
                     Process process = adbOperateUtil.executeDeviceCommandOnPC(deviceInfo, pullComm, logger);
```
Comment (zhou9584) [inline common/src/main/java/com/microsoft/hydralab/common/screen/ADBScreenRecorder.java:102]: please verify the file size after pulling

Comment (olivershen-wow) [reply common/src/main/java/com/microsoft/hydralab/common/screen/ADBScreenRecorder.java]: Updated with retry logic, thx.

File: common/src/main/java/com/microsoft/hydralab/common/util/ADBOperateUtil.java [hunk 2]
```diff
@@ -259,16 +259,43 @@ public void pushFileToDevice(DeviceInfo deviceInfo, String pathOnAgent, String p
     }
 
     public void pullFileToDir(DeviceInfo deviceInfo, String pathOnAgent, String pathOnDevice, Logger logger) throws IOException, InterruptedException {
-        IDevice deviceByInfo = getDeviceByInfo(deviceInfo);
-        Assert.notNull(deviceByInfo, "No such device: " + deviceInfo);
+        final int[] retryTime = {1};
+        FlowUtil.retryWhenFalse(Const.AgentConfig.RETRY_TIME, () -> {
+            logger.info("Pull file round :" + retryTime[0]);
+            File videoFile = new File(pathOnAgent);
```
Comment (zhou9584) [inline common/src/main/java/com/microsoft/hydralab/common/util/ADBOperateUtil.java:265]: Maybe we can replace videoFile with fileOnAgent

**Review by zhou9584 — DISMISSED** [review]

**Review by zhou9584 — APPROVED** [review]

**Review by zhou9584 — APPROVED** [review]

**Review by taoran6 — APPROVED** [review]
LGTM for changes in gradle plugin

**Pull request merged**